### ISSUE TITLE
[Public Catalog] Add the ability to sort quickstarts by relevance, name, and popularity

### ISSUE THREAD

**Issue by zstix** [newrelic/developer-website#1858]
*This is back in to do because we forgot to write tests

## Description
At the moment, there is not currently a way to sort quickstarts on the landing page. The `nr1Catalog` namespace in NerdGraph provides the option to sort search results in a few different ways. Once we have swapped the codebase over to using API calls to power the search & filtering, we should also create the ability to sort.

At the moment, we don't have UI designed for this new sort functionality. We use to have an alphabetical sort set up and we may be able to leverage that code.

_**NOTE**_: while it's possible to sort by popularity via install counts, we can not currently query for the install count itself. Therefore, we should not plan on displaying that information (yet).

_**NOTE**_: at the moment, sorting is one-directional only (no ascending / descending toggle).

#### NerdGraph
Here's the relevant snippet for the NerdGraph query that we would use to handle this:

```graphql
{
  actor {
    nr1Catalog {
      search(sortBy: POPULARITY) {
        results {
          ... on Nr1CatalogQuickstart {
            id
            metadata {
              displayName
            }
          }
        }
      }
    }
  }
}

```

## Acceptance Criteria
* [x] https://github.com/newrelic/developer-website/issues/1853 (landing page via SSR)
* [x] https://github.com/newrelic/developer-website/issues/1855 (search and filter via SSR)
* [x] Create a dropdown for sorting quickstarts
* [x] Sort by relevance
* [x] Sort by name
* [x] Sort by popularity
* [x] Ensure that sorting works while searching and filtering
* [x] Update URL state to reflect current sorting method
* [ ] #1974

**Comment by zstix** [comment]
It's worth noting that `actor.nr1Catalog.search` will return _all_ items in the catalog. If we rely on interface fields (e.g. `... on Nr1CatalogQuickstart`) then we can get what we need, but we will also get a lot of search results that are empty (because they're _not_ quickstarts). This would make it hard for us to get a specific number of `quickstarts`.

After speaking with the Developer Experience team, we're going to look into our options for solving this. Some possible paths we discussed include:
* Adding a `filter` option to `search` that allows us to specify the _type_ (e.g. quickstart vs. nerdpack)
* Adding `sort` and `filter` options to the `quickstarts` namespace

This ticket isn't _blocked_ by this, but we should probably have a plan in place before we start engineering work.

**Comment by aswanson-nr** [comment]
Here's a [link](https://source.datanerd.us/nr1-dev-experience/marketplace/blob/master/common/hooks/useFacets.ts) to how the internal catalog handles the category/component counts using graphql facets. We will want to replicate some of this logic to handle searching/filtering

**Comment by zstix** [comment]
Closing this out. We made a new ticket for writing the tests.
